website/src/blogs/2017-06-29-storage-infrastructure-as-code-using-openebs.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Storage infrastructure as code using OpenEBS
+author: Uma Mukkara
+author_info: Contributor at openebs.io, Co-founder & COO@MayaData. Uma led product development in the early days of MayaData (CloudByte).
+tags: Containerised Storage, DevOps, Kubernetes, Openebs, Stateful Applications
+date: 29-06-2017
+excerpt: With a vision to become de-facto block storage choice for stateful container applications, OpenEBS brings out many unique features to the open source storage space. 
+---
+
+With a vision to become de-facto block storage choice for stateful container applications, OpenEBS brings out many unique features to the open-source storage space. One of the unique features that I talk about in more depth in this blog is “how storage platform builders and storage platform users can manifest their requirements as code using OpenEBS”.
+
+With all the agility in Container and DevOps space, we are moving towards the thinking — “using infrastructure components such as network and storage for the application development should not be external to the application development”. What I mean by this is, just like you write the code for your applications, you should be able to write some code or configuration for obtaining the storage and network infrastructure and you should get it.
+
+#### The YAML: Infrastructure coding language
+
+In the paradigm of infrastructure as code, YAML is becoming the default coding language. Thanks to the increasing usage of YAML config files by the DevOps eco-system, they are becoming the default developer language to write the infrastructure code for the application development.
+
+So, let us see what are the areas in which storage as infrastructure has to be thought in terms of code? This can be answered using the angle of who interacts with storage or the user persona. The simple answer would be the application developer and storage platform engineer. Next, what are the challenges today for a developer of a stateful application and the developer or engineer of underlying persistent storage platform? In short, the developer needs
```

Comment:
```suggestion
So, let us see what are the areas in which storage as infrastructure has to be thought in terms of code? This can be answered using the angle of who interacts with storage or the user persona. The simple answer would be the application developer and storage platform engineer. Next, what are the challenges today for a developer of a stateful application and the developer or engineer of an underlying persistent storage platform? In short, the developer needs
```